utopia-remix/app/services/permissionsService.server.ts

```diff
@@ -0,0 +1,39 @@
+import { assertNever } from '../util/assertNever'
+import { AccessLevel, UserProjectPermission } from '../types'
+import * as fgaService from './fgaService.server'
+
+const ANONYMOUS_USER_ID = '__ANON__'
+
+export async function hasUserProjectPermission(
+  projectId: string,
+  requestUserId: string | null,
+  permission: UserProjectPermission,
+) {
+  const userId = requestUserId ?? ANONYMOUS_USER_ID
```

Comment:
again I think there's no need for this `__ANON__` string. You can just update the `canViewXXX` functions to receive a `null` and return false immediately (it would be good to also add the return type to the function signature)

Reply:
but this is not actually the case, since in public projects we do allow anonymous users to view them. So when a project is confgured so that `user:*` can view it, we need to send to Okta's FGA some user id (even if it's anonymous) for it to return `true`

Reply:
huh ok, I did not know about okta's requiring a user id string